core/src/main/java/hudson/FilePath.java

```diff
@@ -2199,7 +2199,12 @@ public Void invoke(File f, VirtualChannel channel) throws IOException {
                 future.get();
                 return future2.get();
             } catch (ExecutionException e) {
-                throw new IOException(e);
+                Throwable cause = e.getCause();
```

Comment:
It loses the track of `ExecutionException`. I would add it to suppressed at least

Reply:
That is intentional as that is just a wrapper to clearly signify what exceptions are coming from the asynchronous framework and which are thrown by the task itself. We know it is the task that failed and I do not see much value in informing people it was executed asynchronously. The main motivation though was to get rid of the clutter there was a lot of.

Reply:
Hmm, OK. Maybe it would be useful to push more details, e.g. if the FINE logging enabled for the class.